### DISCUSSION TITLE
Need help understanding setDataOnSimObject()

### DISCUSSION THREAD

**Discussion by yourcptainspeaking** [EvenAR/node-simconnect#104] [Q&A] [answered]
I'm trying to use this function to set lvars using the `setDataOnSimObject()` method but I keep getting a `SIMCONNECT_EXCEPTION_INVALID_DATA_SIZE` error. Any idea what I'm doing wrong and how to fix this? Thanks!

```js
public setLVar(name: string, value: number, units: string): void {
	const definitionId = this.generateDefId(2000)

	this.api.handle.addToDataDefinition(
		definitionId,
		`L:${name}`,
		units,
		SimConnectDataType.FLOAT64
	)

	const dataToSet = new RawBuffer(4)
	dataToSet.writeInt32(value)
	this.api.handle.setDataOnSimObject(
		definitionId,
		SimConnectConstants.OBJECT_ID_USER,
		{
			buffer: dataToSet,
			arrayCount: 0,
			tagged: false
		}
	)
}
```

**Comment by EvenAR** [comment, marked as answer]
Hi! I see you specify `FLOAT64` (8 bytes) in the data definition, but when writing to the rawbuffer you use `writeInt32` (4 bytes). This will cause a size mismatch between the `RawBuffer` and the data definition, which could be related to the exception you are getting. Try replacing it with `writeFloat64` (or alternatively use `INT32` in the data definition).

**Reply by yourcptainspeaking** [reply]
Thanks, this worked great! Pretty sure I was thinking of the "L variables are only FLOAT64." note on the SDK docs when I wrote the data definition but forgot about it when doing the buffer. On an unrelated note, is it currently possible to see what specific request caused a simconnect exception?

**Reply by EvenAR** [reply]
Great! 🎉 

> On an unrelated note, is it currently possible to see what specific request caused a simconnect exception?

Yes, it's possible. All SDK function calls returns a send id, and whenever you receive an exception this ID is included [in the payload](https://evenar.github.io/node-simconnect/classes/RecvException.html) (`sendId`) so you can tell which exact function call that caused the error.

**Reply by yourcptainspeaking** [reply]
Perfect, thanks!
